apps/server/src/modules/room/repo/room.repo.ts

```diff
@@ -84,4 +86,20 @@ export class RoomRepo {
 	private async flush(): Promise<void> {
 		return this.em.flush();
 	}
+
+	// belongs into mapper
+	private async mapDoToEntity(room: Room): Promise<RoomEntity> {
+		if (false) {
+			RoomDomainMapper.mapDoToEntity(room);
+		} else {
+			const entity = (await this.em.findOne(RoomEntity, room.id)) || new RoomEntity();
```

Comment:
I think we can use the `UnitOfWork` instead of `await em.findOne()` which would also make the operation non-async, like:
```typescript
const entity = this.em.getUnitOfWork().getById(RoomEntity, room.id);
```
see also:
https://github.com/mikro-orm/mikro-orm/blob/6015f3fb2b09ca920d0fbe3329733d3d5f8d134e/packages/core/src/unit-of-work/UnitOfWork.ts#L128-L145